arch/risc-v/src/common/riscv_addrenv.c

```diff
@@ -539,7 +539,11 @@ int up_addrenv_destroy(arch_addrenv_t *addrenv)
   ptprev = (uintptr_t *)riscv_pgvaddr(addrenv->spgtables[ARCH_SPGTS - 1]);
   if (ptprev)
     {
+  #ifndef CONFIG_ARCH_RV32
```

Comment:
why CONFIG_ARCH_RV32 different from CONFIG_ARCH_RV64? should we unify 32/64bit code path?

Reply:
The logic is correct for RV64 as the walk starts with a level 2 table.
However, for RV32 the walk actuallly starts with the root table where PTE 0 means vaddr 0, so it needs be adjusted so that the walk applies only to the user space (and skip the kernel space).